### PR TITLE
[bitnami/kafka] Add networkpolicy support

### PR DESCRIPTION
**Pull request by bbaudelet**
**Description of the change**

Add networkpolicy support on kafka chart.
Networkpolicy control access to Kafka port for client connections, if metrics is enabled flow are not filtered and open from any source. communication inter kafka pod is allowed. 

**Benefits**

If the chart is deployed on a K8S cluster where default rule is deny all, we need to deployed a networkpolicy but it can be tedious. we want to have a chart allign with other ones to integrate a default netwokpolicy.

**Possible drawbacks**

By default network policy is disable, but when it's enabled some change must be done to be work fine.
the good label and sometime the right namespaces selector must be added to each pod, want to reach the kafka cluster.

**Applicable issues**

  - fixes #

**Additional information**

**Checklist** 

- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. [bitnami/<name_of_the_chart>])

### CODE REVIEW

**Comment by madrisan** [comment]
Great PR. May you also add the support for `service.externalPort`?

File: bitnami/kafka/README.md [hunk 1]
```diff
@@ -146,61 +146,63 @@ The command removes all the Kubernetes components associated with the chart and
 | `auth.tlsEndpointIdentificationAlgorithm`  | DEPRECATED: use `auth.tls.endpointIdentificationAlgorithm` instead.                                                                                  | `https`                             |
 | `listeners`                                | The address(es) the socket server listens on. Auto-calculated it's set to an empty array                                                             | `[]`                                |
 | `advertisedListeners`                      | The address(es) (hostname:port) the broker will advertise to producers and consumers. Auto-calculated it's set to an empty array                     | `[]`                                |
-| `listenerSecurityProtocolMap`              | The protocol->listener mapping. Auto-calculated it's set to nil                                                                                      | `""`                                |
+| `listenerCProtocolMap`              | The protocol->listener mapping. Auto-calculated it's set to nil                                                                                      | `""`                                |
```
Comment (juan131) [inline bitnami/kafka/README.md:149]: Why did you change this? I mean, the parameter hasn't been renamed.

File: bitnami/kafka/values.yaml [hunk 2]
```diff
@@ -612,6 +612,31 @@ sidecars: []
...
+  ## LabelSelector to select these namespaces, note that the networkPolicy's namespace should also be explicitly added.
+  ##
+  ## Example:
+  ## explicitNamespacesSelector:
+  ##   matchLabels:
+  ##     role: frontend
+  ##   matchExpressions:
+  ##    - {key: role, operator: In, values: [frontend]}
+  ##
+  explicitNamespacesSelector: {}
```
Comment (juan131) [inline bitnami/kafka/values.yaml:639]: I'd include all these parameters under the section below "Exposure parameters"

File: bitnami/kafka/templates/networkpolicy.yaml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+              {{ template "common.names.fullname" . }}-client: "true"
+          {{- if .Values.networkPolicy.explicitNamespacesSelector }}
+          namespaceSelector:
+{{ toYaml .Values.networkPolicy.explicitNamespacesSelector | indent 12 }}
+          {{- end }}
+    # Allow communication inter-broker
+        - podSelector:
+            matchLabels:
+            {{- include "common.labels.matchLabels" . | nindent 14 }}
+      {{- end }}
```
Comment (juan131) [inline bitnami/kafka/templates/networkpolicy.yaml:34]: Did you test this rule? Inter-broker communications use a different port (obtained from `.Values.service.internalPort` instead of `.Values.service.port`), therefore I'd say you're not allowing inter-broker traffic.

File: bitnami/kafka/values.yaml (hunk 2 above)
Comment (juan131) [reply bitnami/kafka/values.yaml:639]: Use the [readme-generator](https://github.com/bitnami-labs/readme-generator-for-helm) tool then to update the README.md table to reflect the changes.

File: bitnami/kafka/Chart.yaml [hunk 4]
```diff
@@ -29,4 +29,4 @@ name: kafka
 sources:
   - https://github.com/bitnami/bitnami-docker-kafka
   - https://kafka.apache.org/
-version: 14.5.0
+version: 14.5.1
```
Comment (juan131) [inline bitnami/kafka/Chart.yaml:32]: Please bump the minor version instead:
```suggestion
version: 14.6.0
```

**Review by juan131 — CHANGES_REQUESTED** [review]
Thanks so much for this great contribution!! Please check my comments when you have a chance!

File: bitnami/kafka/README.md (hunk 1 above)
Comment (bbaudelet) [reply bitnami/kafka/README.md]: yes you are right, it's my bad sorry.

File: bitnami/kafka/values.yaml (hunk 2 above)
Comment (juan131) [reply bitnami/kafka/values.yaml:639]: This is not done, right?

File: bitnami/kafka/templates/networkpolicy.yaml [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+        - podSelector:
+            matchLabels:
+            {{- include "common.labels.matchLabels" . | nindent 14 }}
+    # Allow External connection
+    {{- if .Values.externalAccess.enabled }}
+    - ports: {{ .Values.service.externalPort }}
+      from:
+        - podSelector:
+            matchLabels:
+              {{ template "common.names.fullname" . }}-external: "true"
```
Comment (juan131) [inline bitnami/kafka/templates/networkpolicy.yaml:43]: External connections do not come from any pod. Why do you need this?

File: bitnami/kafka/values.yaml (hunk 2 above)
Comment (bbaudelet) [reply bitnami/kafka/values.yaml:640]: hi, i have just move networkPolicy paramerters under Exposure section and i have re generate readme.

File: bitnami/kafka/templates/networkpolicy.yaml (hunk 5 above)
Comment (bbaudelet) [reply bitnami/kafka/templates/networkpolicy.yaml]: yes you are right, i have updated this section to let the user inject the from parameters to use no filtering (if not defined), ipBlock or other configuration.

File: bitnami/kafka/templates/networkpolicy.yaml [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+    {{- end }}
+    {{- end }}
+    {{- if .Values.metrics.kafka.enabled }}
+    # Allow prometheus scrapes
+    - ports:
+        - port: {{ .Values.metrics.kafka.service.port }}
+    {{- end }}
+{{- end }}
+
+
```
Comment (juan131) [inline bitnami/kafka/templates/networkpolicy.yaml:54]:
```suggestion
```

**Review by juan131 — APPROVED** [review]
LGTM! Thanks for including the suggestions

**Comment by juan131** [comment]
@bbaudelet the CI is complaining about the chart version.. Could you please update your fork with the latest changes in master and rebase your branch? Thanks in advance.

**Comment by juan131** [comment]
Thanks @bbaudelet
